packages/core/src/types/builder-options.ts

```diff
@@ -175,14 +180,14 @@ export type InterfaceTypeOptions<
 export type EnumTypeOptions<
   Types extends SchemaTypes,
   Param extends EnumParam,
-  Values extends EnumValues<Types>,
+  Values extends EnumValues<Types, Param>,
 > = Param extends BaseEnum
   ? Merge<
-      Omit<PothosSchemaTypes.EnumTypeOptions<Types, Values>, 'values'> & {
+      PartialBy<PothosSchemaTypes.EnumTypeOptions<Types, Param, Values>, 'values'> & {
```

Comment:
rather than trying to hack around this special case, it will be a lot easier just to leave all the existing type as is, and add in values on the line below.

Basically, you should be able to add a new field like `values?: Partial<Record<KeysFromParam<Param>, Omit<EnumValueOptions, 'value'>>`  on the line below, along side name.  That object already is customizing the options for the case where the Param was an Enum value, and the normal values option has been removed, so you can add a new values option specific to this case.

The type in the comment above is awful sudo-code and not actually how that would be implemented, but something similar should work.

Reply:
doing it this way should avoid needing to change any of the existing shared types, so that this doesn't break existing usages of those types.